street_network/src/types.rs

```diff
@@ -124,12 +124,64 @@ pub enum DrivingSide {
     Left,
 }
 
+/// How a lane of travel is interrupted.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub enum IntersectionType {
-    StopSign,
-    TrafficSignal,
-    Border,
-    Construction,
+pub enum InterruptionType {
+    Uninterrupted,
+    Yield,
+    Stop,
+    Signal,
+    DeadEnd,
+}
```

Comment:
This is not used yet, so I will probably delete it for this PR. Both end of every road should be classified with one of these, I think.

Reply:
I like this. End of every _lane_ though, right?

Reply:
Yes. Every *manoeuvre* I suppose. Lanes divvy themselves up, by tags and defaults.

I mean, any time you're turning across traffic, you pull up and give way at some point. Every driveway is a give way for any traffic doing that turn manoeuvre (but private driveways tend to be ignored, unless you see something). So the same lane is both uninterrupted and sometimes give way. (if the person in front of you is giving way, so are you, as long as you stay in the lane). Traffic lights are also uninterrupted and sometimes stop, I guess.

Here in Aus, we have double white lines, for no overtaking on long highways. You can drive straight ahead as long as you like, without and conflict, or turn into a parking bay or side road. Because you can't cross double white lines, every driveway they you're allowed to use across traffic has a small gap in the doubles, to show that you're allowed across. I'll see if I can find a picture somewhere.

Reply:
[Proper gap at a Yielding road](https://www.mapillary.com/app/?lat=-34.970524391611&lng=116.94807567397&z=17&pKey=3894113797324472&focus=photo) then a [tiny one at a driveway](https://www.mapillary.com/app/?lat=-34.9700375617&lng=116.94865674417&z=17&pKey=688989171843342&focus=photo) just up the road.